vllm/worker/hpu_model_runner.py

```diff
@@ -309,18 +310,20 @@ def _set_attn_bias(self, attn_metadata, batch_size, seq_len, device,
         attn_metadata = prefill_metadata._replace(attn_bias=attn_bias)
         return attn_metadata
 
-    def _set_block_mapping(self, metadata, batch_size, device, dtype):
+    def _set_block_mapping(self, metadata, batch_size, device, dtype,
+                           warmup_mode):
         mask = torch.arange(0,
                             self.block_size,
                             device=device,
                             dtype=torch.int32).unsqueeze(0)
         mask = mask >= metadata.block_usage.unsqueeze(-1)
         attn_bias = (torch.zeros_like(mask, dtype=dtype).masked_fill_(
             mask, -math.inf))
-        if is_fake_hpu():
-            # Unfortunately one_hot on CPU doesn't handle
-            # out of bounds classes. We need to mask those
-            # values manually
+        if is_fake_hpu() or (not htorch.utils.internal.is_lazy()
+                             and not self.enforce_eager and not warmup_mode):
+            # Unfortunately one_hot on CPU or in torch compile
+            # mode doesn't handle out of bounds classes.
+            # We need to mask those values manually
             oob_values = metadata.block_mapping.lt(0)
             block_mapping = metadata.block_mapping.masked_fill(oob_values, 0)
```

Comment:
maybe torch.relu() will be good enough and faster than creating mask ?

Reply:
Great idea! I switched to using relu() instead.